platform/engineering/request-for-change/2019-09-30-cms-integration.md

```diff
@@ -0,0 +1,230 @@
+`vets-website`.
+
+### Content validation
+Some basic validations are performed on the content from the CMS and
+vagov-content repo.
+
+| Validation | Full build | Content-only build | Preview server |
+| ---------- | ---------- | ------------------ | -------------- |
+| Broken link checking | True | True | False |
+| Accessibility | True | True | False |
+
+The Metalsmith script performs the check for broken internal
+links. This happens before the accessibility tests in the Jenkins
+build pipelines.
+
+When broken links are discovered during a full Jenkins build, we send
+a Slack notification with a link to the build. If Jenkins was building
+the production (`master`) branch, the build will also fail, to prevent
+broken links in production.
+
+Accessibility tests run the aXe checker on all pages found in the
+sitemap. Whenever the accessibility check fails, the Jenkins build
+fails. This happens after the content build and link checker.
+
+## Motivation
+_Why do we want to change the current implementation? What problem(s)
+does the change solve?_
+
+When any new commits are pushed to a `vets-website` branch, a full
+Jenkins build is triggered. This build attempts to fetch the latest
+content and do some content validation (see [Content
+Validation](#content-validation) for details). This means that **two
+subsequent builds can fail for reasons not introduced in the code.**
+
+The biggest example of this is when the `master` branch build fails
+due to broken links, which will halt the deploy. **Broken links
+prevent unrelated features in React applications from being
+deployed.**
+
+**Example**
+1. Somebody merges a branch into `master`
+1. A full Jenkins build is triggered, which then passes
+1. A broken link is introduced in the CMS
+1. Somebody else merges a branch into `master`
+1. A new full Jenkins build is triggered
+1. The Metalsmith script queries for the latest content with the
+   broken link
+1. Because of this latest content, the `vets-website` `master` build
+   fails, but not because of any changes in `vets-website` code
+
+The goal of this RFC is to determine an approach for separating the
+content build from the `vets-website` Webpack build, so when
+`vets-website` pulls the content to serve up, it's already been built
+and validated. **Build failures due to content issues will be isolated
+to the content build and will not affect `vets-website` builds or
+deploys.** This will improve the overall stability of the
+`vets-website` CI.
+
+## Design
+_Explain the proposed changes in enough detail so that a team member
+with working knowledge of the project could implement the change
+themselves. Bonus points if this is easily mappable to user stories or
+specs._
+
+I propose rearchitecting the `vets-website` build to separate the
+static content building into a process distinct from the website.
+
+![separate build diagram](images/separated-content-build-proposal.png)
```

Comment:
if it's not too much effort to do,  this diagram might be a lot clearer with a similar "current build process" diagram, to visually see what's changing

Reply:
Done! I put it in the **Background** section, though. I can move it if desired.